pandas/core/internals/ops.py

```diff
@@ -34,21 +31,31 @@ def operate_blockwise(
             right_ea = not isinstance(rblk.values, np.ndarray)
 
             lvals, rvals = _get_same_shape_values(blk, rblk, left_ea, right_ea)
+            yield lvals, rvals, locs, left_ea, right_ea, rblk
```

Comment:
maybe yield a NamedTuple?

Reply:
IIRC there are issues with NamedTuple construction performance

Reply:
maybe wait a couple of days and use a dataclass.

Reply:
> IIRC there are issues with NamedTuple construction performance

really? can you point to something that shows this

Reply:
May be out of date but I think this is where i got the impression that NamedTuple has perf issues: https://lwn.net/Articles/731423/

Reply:
i prefer readability here
unless it's way slower